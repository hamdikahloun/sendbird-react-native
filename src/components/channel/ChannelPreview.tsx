import React, {memo} from 'react';
import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {OpenChannel} from 'sendbird';
import {Avatar} from '@/components';

type ChannelProps = {
  channel: OpenChannel;
  onChannelPressed?: (channel: OpenChannel) => void;
};
const ChannelPreviewMemorized = ({channel, onChannelPressed}: ChannelProps) => {
  const onPress = () => {
    onChannelPressed?.(channel);
  };
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Avatar image={channel.coverUrl} />
      <View style={styles.wrapper}>
        <Text style={styles.name}>{channel.name}</Text>
        <Text style={styles.subtitle}>
          {channel.participantCount + ' Participant'}
        </Text>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    paddingHorizontal: 10,
  },
  button: {
    padding: 10,
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#F9F9F9',
    borderRadius: 15,
    borderWidth: 1,
    borderColor: 'white',
  },
  name: {
    fontSize: 16,
    color: '#0C0020',
  },
  subtitle: {
    color: '#0C0020',
    fontSize: 14,
  },
  wrapper: {
    marginStart: 16,
  },
});

/**
 * ChannelPreview - This UI component displays an individual preview item for each channel in a list
 *
 * See:
 * [Channel](https://sendbird.com/docs/chat/v3/platform-api/guides/channel-types)
 * @example
 *
 * <ChannelPreview onChannelPressed={onChannelPressed} channel={channel} />
 */
export const ChannelPreview = memo(ChannelPreviewMemorized);
