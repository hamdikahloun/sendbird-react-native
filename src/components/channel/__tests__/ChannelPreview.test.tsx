import React from 'react';
import {render} from '@/../testing';
import {ChannelPreview} from '../ChannelPreview';

describe('ChannelPreview', () => {
  it('should render correctly', async () => {
    const channel: any = {coverUrl: 'https://example.com', name: 'My Channel'};
    const screen = render(<ChannelPreview channel={channel} />).toJSON();
    expect(screen).toMatchSnapshot();
  });
});
