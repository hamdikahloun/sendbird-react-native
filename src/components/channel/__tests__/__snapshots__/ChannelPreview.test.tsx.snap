// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChannelPreview should render correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F9F9F9",
      "borderColor": "white",
      "borderRadius": 15,
      "borderWidth": 1,
      "flexDirection": "row",
      "opacity": 1,
      "padding": 10,
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 24,
            "height": 48,
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <Image
        onError={[Function]}
        source={
          Object {
            "uri": "https://example.com",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 24,
              "height": 48,
              "width": 48,
            },
            undefined,
          ]
        }
        testID="user-avatar"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginStart": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0C0020",
          "fontSize": 16,
        }
      }
    >
      My Channel
    </Text>
    <Text
      style={
        Object {
          "color": "#0C0020",
          "fontSize": 14,
        }
      }
    >
      undefined Participant
    </Text>
  </View>
</View>
`;
